### ISSUE TITLE
"Execute TP" command Fails in Command Block in Forge (and not Vanilla)

### ISSUE THREAD

**Issue by mactso** [MinecraftForge/MinecraftForge#8262]
**Minecraft Version:** {Minecraft version}
Minecraft 1.18  (and also confirmed in 1.17 and 1.16.5)
**Forge Version:** {Forge version. *Version number, not latest/rb*}
38.0.16 ( and also confirmed in older forge versions 37, 36 so this issue wasn't created going to 1.18)
**Logs:** {Link(s) to GitHub Gist with full latest.log and/or crash report}

**Steps to Reproduce:**
**Overworld Procedure**
 1. While in the overworld Execute this command manually
 /execute in minecraft:the_nether run teleport @p 0 128 0
 2.  Result: You are teleported to location 0 128 0 in the overworld, the game is running normally.
 3. Then Execute the same command in a command block
 4. Result: Game is running on a bluish grey screen and nothing is rendered.  It is not hung.
 5. Pressing the F3 command works (showing the game isn't hung)
 6. You can recover by executing the teleport command manually.
**Nether Procedure**
 1. While in the nether execute this command manually
 /execute in minecraft:overworld run teleport @p 0 128 0
 2. Result: You are teleported to location 0 128 0 in the overworld, the game is running normally.
 3. Then Execute the same command in a command block in the nether.
 4. Result: Game is running on a greenish grey screen and nothing is rendered.  But It is not hung.
 5. Pressing the F3 command works (showing the game isn't hung)
 6. You can recover by executing the teleport command manually.

**Description of issue:**
Running an "execute teleport in anotherdimension" command in FORGE in a command block to teleport to a different dimension puts the game into a running (not hung) state where the world is not being rendered.  In vanilla, executing a teleport command in a different dimension works fine.

**Comment by mactso** [comment]
We have verified this was working correctly in 1.16.2.

**Comment by mactso** [comment]
Okay, we continued to track through this in verison 16 and it looks like it may have first shown up in version 17 in september.    (which seemed like a million years ago).  We are confirming 1.16.5 now.  We suspect it will be okay.  That will mean this is a minecraft 17/18 bug introduced in 17.

**Comment by mactso** [comment]
Okay we have confirmed this was not an issue in version 16.5.   It started in  Version 17 and continues into 18.  Please remove the 16 tag.

With Best Regards,
Two Guys,
Mactso & Lupicus

**Comment by TheCurle** [comment]
Very interesting!
After stepping through with a debugger, it seems like when the command is run from a command block, it encounters an edge case, which throws an exception and leaves the player in a preparing-to-teleport but not actually teleported state:
![image](https://user-images.githubusercontent.com/42079760/145320458-f5f6d711-ba9a-4c8a-9b74-627cc2e1d2bc.png)

**Comment by TheCurle** [comment]
I've narrowed down the cause of the fault to this patched-in method:
https://github.com/MinecraftForge/MinecraftForge/blob/dcf7bb1af28233840db529222cc2c62a75cf151b/patches/minecraft/net/minecraft/server/level/ServerPlayer.java.patch#L33..L38

```
-         serverlevel.m_143261_(this, Entity.RemovalReason.CHANGED_DIMENSION);
+         serverlevel.removePlayer(this, true); //Forge: The player entity itself is moved, and not cloned. So we need to keep the data alive with no matching invalidate call later.
```

Looking into this some more.

**Comment by robotgryphon** [comment]
+1 per player reports on 1.18.1, `execute as @p in minecraft:the_nether run tp @s 1033.7 40.0 1032` fails on a base Forge install.

![image](https://user-images.githubusercontent.com/8529636/148631640-a88e40e2-d1ea-4f96-9b89-926a9235e587.png)

**Comment by LiskoSlayer63** [comment]
I found another way to make this happen. Steps to reproduce:

1. Install Forge (39.0.58 to be exact)
2. Install [Nullscape](https://www.curseforge.com/minecraft/mc-mods/nullscape-end-reborn)
3. Go to The End
4. Kill the Dragon
5. Return from The End
6. Try to enter the portal again and this happens; stuck in the void, can't go to pause menu and log spams: 
```
LiskoSlayer63 moved too quickly! 2014.6484875327844,-2.8431999969482433,-7072.671815435907
Skipping update from removed player 'ServerPlayer['LiskoSlayer63'/256, l='ServerLevel[world]', x=-2006.15, y=67.84, z=7081.17, removed=DISCARDED]'
```

***UPDATE:***
Tried this #8434 PR and it fixes the problem. Is there some reasons why it isn't merged yet to the official repo?

***UPDATE 2:***
It didn't fully solve the problem, now it happens randomly but it spams the message a lot slower. When that happens, the server crashes because of the watchdog kills it after 60secs.
